src/openforms/js/components/admin/form_design/logic/Trigger.js

```diff
@@ -293,6 +294,14 @@ const Trigger = ({name, logic, onChange, error, withDSLPreview = false, children
       default:
         firstOperand = {var: triggerVariableKey};
     }
+  } else if (
+    triggerVariable?.source === VARIABLE_SOURCES.userDefined &&
+    triggerVariable.dataType === 'boolean'
+  ) {
+    firstOperand = {var: triggerVariableKey};
+    // cast from string to actual boolean
+    if (compareValue === 'true') compareValue = true;
+    if (compareValue === 'false') compareValue = false;
```

Comment:
```suggestion
    compareValue = STRING_TO_TYPE.bool(compareValue);
```

Or if you prefer to keep it this way, the import is unused